### DISCUSSION TITLE
BRATS tutorial - Test custom NIFTI images

### DISCUSSION THREAD

**Discussion by sekh77** [Project-MONAI/MONAI#1399] [Q&A] [answered]
Hi -

I'm using the latest BRATS tutorial on MONAI v0.4.0. I have a couple of NIFTI images in my data folder very much similar to the Decathlon Dataset. How do I use these images to test the model? The provided code tutorial in github, currently, selects one image from the validation dataset to evaluate and visualize the model output [referring to this line of the code: _val_input = val_ds[10]["image"].unsqueeze(0).to(device)_].

Instead of the Decathlon validation dataset, I will like to use my dataset. Please advise the changes that I need to do to test the images.

Thanks,
Sekhar H.

**Comment by architraj29** [comment, marked as answer]
Hi Shekhar,


Following piece of code is used in [Brats tutorial](https://github.com/Project-MONAI/tutorials/blob/0.3.0/modules/integrate_3rd_party_transforms.ipynb) to specify the validation data:
```
val_ds = DecathlonDataset(
    root_dir=root_dir,
    task="Task01_BrainTumour",
    transform=val_transform,
    section="validation",
    download=False,
    num_workers=4,
)
val_loader = DataLoader(val_ds, batch_size=2, shuffle=False, num_workers=4)
```

Replace the above code with the following (this was you can specify the exact files in a folder that need to be evaluated):

Specify:
- root data path
- data loading type (Detailed analysis could be found [here](https://github.com/Project-MONAI/tutorials/blob/0.3.0/acceleration/dataset_type_performance.ipynb))
```
root_data_path = '<Base path for your Validation Datasets>'
val_images = sorted(glob.glob(os.path.join(root_data_path, 'imagesTs', '*.nii.gz')))
val_labels = sorted(glob.glob(os.path.join(root_data_path, 'labelsTs', '*.nii.gz')))
val_files = [{"image": img, "label": seg} for img, seg in zip(val_images, val_labels)]

# Specify the loading type depending upon system memory
# Cache is fastest but takes up more memory
# Persistent is almost as fast as cache
# Default is the slowest 

loading_type = 'persistent'

if loading_type == 'persistent':
    persistent_cache = pathlib.Path(self.root_data_path, "persistent_cache")
    persistent_cache.mkdir(parents=True, exist_ok=True)
    val_ds = monai.data.PersistentDataset(data=val_files, transform=val_transforms, cache_dir=persistent_cache)
    
elif loading_type == 'cache':
    val_ds = monai.data.CacheDataset(data=val_files, transform=val_transforms, cache_rate=1.0)
    
else:
    val_ds = monai.data.Dataset(data=val_files, transform=val_transforms)
    
val_loader = monai.data.DataLoader(val_ds, batch_size=self.val_batch_size, num_workers=4)
```

Thanks,
Archit
